### PR TITLE
Run DPE verification tests against Caliptra

### PR DESCRIPTION
**Pull request by jhand2**
Use HwModel and libcaliptra to run DPE's verification tests (written in Go) against Caliptra.

This suite tests conformance with the DPE spec as well as related X.509 and PKCS#10 specs.

### CODE REVIEW

File: test/dpe_verification/transport.go [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+	"io/ioutil"
+	"log"
+	"os"
+	"unsafe"
+
+	"github.com/chipsalliance/caliptra-dpe/verification"
+)
+
+type CptraModel struct {
+	exe_path        string
```
Comment (jhand2) [inline test/dpe_verification/transport.go:61]: This isn't necessary. Caliptra HwModel doesn't execute a subprocess

File: test/dpe_verification/transport.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+// Kill the emulator in a way that it can cleanup before closing.
+func (s *CptraModel) PowerOff() error {
+	C.caliptra_model_destroy(CALIPTRA_C_MODEL)
+	CALIPTRA_C_MODEL = nil
+
+	return nil
+}
+
+func fileExists(filename string) bool {
```
Comment (jhand2) [inline test/dpe_verification/transport.go:162]: Unused. Remove

File: test/dpe_verification/transport.go [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+// Wait for the emulator to come alive. Timeout at 15 seconds.
+func (s *CptraModel) waitForPower(on bool) bool {
```
Comment (jhand2) [inline test/dpe_verification/transport.go:171]: Unused. HwModel startup will block until fw is running.

**Comment by jhand2** [comment]
Fix for rt panic CI failure is at https://github.com/chipsalliance/caliptra-dpe/pull/236

File: runtime/src/dpe_platform.rs [hunk 4]
```diff
@@ -80,12 +80,13 @@ impl Platform for DpePlatform<'_> {
         const CALIPTRA_CN: &[u8] = b"Caliptra Rt Alias";
         let mut issuer_writer = X509CertWriter::new(out, true);
 
-        let mut serial = [0u8; DPE_PROFILE.get_hash_size() * 2];
+        // Caliptra RDN SerialNumber fiel is always a Sha256 hash
```
Comment (sree-revoori1) [inline runtime/src/dpe_platform.rs:83]: nit: field

File: runtime/src/drivers.rs [hunk 5]
```diff
@@ -228,21 +228,29 @@ impl Drivers {
         Ok(())
     }
 
+    // Caliptra Name serialNumber fields are sha256 digests
+    pub fn compute_rt_alias_sn(&mut self) -> CaliptraResult<CryptoBuf> {
+        let key = self.persistent_data.get().fht.rt_dice_pub_key.to_der();
+
+        let rt_digest = self.sha256.digest(&key)?;
```
Comment (sree-revoori1) [inline runtime/src/drivers.rs:235]: should we skip hashing the 0x04 der encoding byte?

Comment (jhand2) [reply runtime/src/drivers.rs:235]: FMC hashes it and we have to do the same thing as FMC. Anyway, hashing the uncompressed format of the public key is a reasonable standard to follow

Comment (sree-revoori1) [reply runtime/src/drivers.rs:235]: SGTM

**Review by bluegate010 — DISMISSED** [review]

File: runtime/src/dpe_platform.rs (hunk 4 above)
Comment (jhand2) [reply runtime/src/dpe_platform.rs]: Done

File: test/dpe_verification/transport.go [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+func (s *CptraModel) SetLocality(locality uint32) {
+	// TODO: Hook this up to HwModel
+	s.currentLocality = locality
+}
+
+func (s *CptraModel) GetLocality() uint32 {
+	return s.currentLocality
+}
+
+func (s *CptraModel) GetMaxTciNodes() uint32 {
```
Comment (sree-revoori1) [inline test/dpe_verification/transport.go:221]: Not too familiar with go/rust integration, so not sure if this is possible but is there a way we can import these constants from dpe/src/lib.rs?

**Review by sree-revoori1 — APPROVED** [review]

File: test/dpe_verification/transport.go (hunk 6 above)
Comment (jhand2) [reply test/dpe_verification/transport.go:221]: Technically yes

1. In build.rs, generate C headers with cbindgen
2. Include these headers in cgo
3. Use the constants with C.*

I'll leave this for now and we can maybe look into doing this in the future.

**Review by wmaroneAMD — APPROVED** [review]

**Review by ericeilertson — APPROVED** [review]

**Review by bluegate010 — APPROVED** [review]

File: libcaliptra/Makefile [hunk 7]
```diff
@@ -3,6 +3,8 @@ Q=@
 CC=$(CROSS_COMPILE)gcc
 AR=$(CROSS_COMPILE)ar
 
+RTL_SOC_IFC_INCLUDE_PATH ?= ../hw-latest/caliptra-rtl/src/soc_ifc/rtl
```
[the remark this answers is not in the document]
Comment (jhand2) [reply libcaliptra/Makefile:6]: Will follow up on this and other comment in another PR

**Pull request merged**